custompuppet.go

```diff
@@ -66,7 +66,10 @@ func (br *SlackBridge) newDoublePuppetClient(mxid id.UserID, accessToken string)
 		return nil, err
 	}
 
-	client.Logger = br.AS.Log.Sub(mxid.String())
+	client.Log = br.AS.Log.With().
+		Str("component", "custompuppet").
+		Str("mxid", mxid.String()).
+		Logger()
 	client.Client = br.AS.HTTPClient
 	client.DefaultHTTPRetries = br.AS.DefaultHTTPRetries
 
```

Comment:
This whole thing and the two lines above and below should just be `return br.AS.NewExternalMautrixClient(mxid, accessToken, homeserverURL)`